sktime/transformations/series/summarize.py

```diff
@@ -750,3 +750,140 @@ def get_test_params(cls, parameter_set="default"):
         params4 = {"summary_function": None, "quantiles": (0.1, 0.2, 0.25)}
 
         return [params1, params2, params3, params4]
+
+
+class SplitterSummarizer(BaseTransformer):
+    """Create summary values of a time series' splits.
+
+    A series-to-series transformer that applies a series-to-primitives transformer
+    to each series' train split.
+
+    It splits the series using a specified splitter and transforms each resulting
+    split using the provided transformer.
+
+    This transformer aims to provide a summarization of the series based on the
+    transformer and splitter given.
+
+    Attributes
+    ----------
+    transformer : series-to-primitives transformer used to convert series
+        to primitives.
+
+    splitter : splitter used to divide the series.
+
+    Methods
+    -------
+    transform(X) : Transforms the series according to the specified
+        series-to-primitives transformer and splitter.
+
+    See Also
+    --------
+    SummaryTransformer: Calculates summary value of a time series.
+
+    Notes
+    -----
+    Empty
+
+    Examples
+    --------
+    >>> from sktime.transformations.series.summarize import SplitterSummarizer
+    >>> from sktime.transformations.series.summarize import SummaryTransformer
+    >>> from sktime.forecasting.model_selection import ExpandingWindowSplitter
+    >>> from sktime.datasets import load_airline
+    >>> y = load_airline()
+    >>> transformer = SplitterSummarizer(
+    >>>     transformer=SummaryTransformer(),
+    >>>     splitter=ExpandingWindowSplitter())
+    >>> y_splitsummarized = transformer.fit_transform(y)
+    """
+
+    _tags = {
+        "scitype:transform-input": "Series",
+        # what is the scitype of X: Series, or Panel
+        "scitype:transform-output": "Series",
+        # what scitype is returned: Primitives, Series, Panel
+        "scitype:instancewise": True,  # is this an instance-wise transform?
+        "X_inner_mtype": ["pd.DataFrame", "pd.Series"],
+        # which mtypes do _fit/_predict support for X?
+        "y_inner_mtype": "None",  # which mtypes do _fit/_predict support for X?
+        "fit_is_empty": True,
+    }
+
+    def __init__(self, transformer, splitter, index="last"):
+        self.transformer = transformer
+        self.splitter = splitter
+        self.index = index
+
+        super().__init__()
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        Transforms the series `X` into a new series `Xt` by applying the
+        series-to-primitives transformer `t` on each train split of `X`,
+         which is created using the splitter `cv`.
+
+         The `i-th` row of `Xt` is equivalent to
```

Comment:
This should be in the docstring of the *class*, this is good!

The docstrings of the private methods are not displayed in the API docs so not visible to the user.

Reply:
I moved it to class' docstring